chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/endpoints/ChaosMonkeyRestEndpoint.java

```diff
@@ -72,14 +76,26 @@ public AssaultPropertiesUpdate getAssaultSettings() {
 
   @PostMapping("/enable")
   public ResponseEntity<ChaosMonkeyEnabledDto> enableChaosMonkey() {
-    this.chaosMonkeySettings.getChaosMonkeyProperties().setEnabled(true);
-    return ResponseEntity.ok().body(new ChaosMonkeyEnabledDto());
+    if (!this.chaosMonkeySettings.getChaosMonkeyProperties().isEnabled()
+        || chaosMonkeyEnabledDto == null) {
+      chaosMonkeyEnabledDto = new ChaosMonkeyEnabledDto();
+      this.chaosMonkeySettings.getChaosMonkeyProperties().setEnabled(true);
+    }
+    return ResponseEntity.ok().body(chaosMonkeyEnabledDto);
   }
 
   @PostMapping("/disable")
   public ResponseEntity<ChaosMonkeyDisabledDto> disableChaosMonkey() {
-    this.chaosMonkeySettings.getChaosMonkeyProperties().setEnabled(false);
-    return ResponseEntity.ok().body(new ChaosMonkeyDisabledDto());
+    if (this.chaosMonkeySettings.getChaosMonkeyProperties().isEnabled()
+        || chaosMonkeyDisabledDto == null) {
+      if (chaosMonkeyEnabledDto == null) {
+        chaosMonkeyDisabledDto = new ChaosMonkeyDisabledDto();
+      } else {
+        chaosMonkeyDisabledDto = new ChaosMonkeyDisabledDto(chaosMonkeyEnabledDto);
+      }
+      this.chaosMonkeySettings.getChaosMonkeyProperties().setEnabled(false);
+    }
+    return ResponseEntity.ok().body(chaosMonkeyDisabledDto);
```

Comment:
I think as soon you store the enabled timestamp somewhere else, this could be simplified